packages/client/src/views/RecordAudit/ActionMenu.tsx

```diff
@@ -0,0 +1,646 @@
+          />
+          <ArchiveAction
+            toggleDisplayDialog={toggleDisplayDialog}
+            isDownloaded={isDownloaded}
+            intl={intl}
+            scope={scope}
+            declarationStatus={status}
+          />
+          <ReinstateAction
+            toggleDisplayDialog={toggleDisplayDialog}
+            isDownloaded={isDownloaded}
+            intl={intl}
+            scope={scope}
+            declarationStatus={status}
+          />
+          <PrintAction
+            declarationStatus={status}
+            declarationId={id}
+            type={type}
+            isDownloaded={isDownloaded}
+            intl={intl}
+            scope={scope}
+            clearCorrectionAndPrintChanges={clearCorrectionAndPrintChanges}
+            goToPrintCertificate={goToPrintCertificate}
+          />
+          <IssueAction
+            declarationStatus={status}
+            declarationId={id}
+            isDownloaded={isDownloaded}
+            intl={intl}
+            scope={scope}
+          />
+          <CorrectRecordAction
+            declarationId={id}
+            declarationStatus={status}
+            type={type}
+            isDownloaded={isDownloaded}
+            intl={intl}
+            scope={scope}
+          />
+          <DeleteAction
+            handleDelete={handleDelete}
+            intl={intl}
+            declarationStatus={status}
+          />
+          <UnassignAction
+            handleUnassign={handleUnassign}
+            intl={intl}
+            isDownloaded={isDownloaded}
+            scope={scope}
+            assignment={assignment}
+          />
+        </DropdownMenu.Content>
+      </DropdownMenu>
+      {modal}
+    </>
+  )
+}
+
+interface IActionItemCommonProps {
+  isDownloaded: boolean
+  intl: IntlShape
+  scope: Scope
+  declarationStatus?: string
+}
+
+interface IDeclarationProps {
+  declarationId: string
+  type?: string
+}
+
+const ViewAction: React.FC<{
+  intl: IntlShape
+  declarationStatus?: string
+  declarationId: string
+}> = ({ declarationStatus, declarationId, intl }) => {
+  const dispatch = useDispatch()
+
+  const recordOrDeclaration = [
+    SUBMISSION_STATUS.REGISTERED,
+    SUBMISSION_STATUS.CORRECTION_REQUESTED,
+    SUBMISSION_STATUS.CERTIFIED
+  ].includes(declarationStatus as any as SUBMISSION_STATUS)
+    ? 'record'
+    : 'declaration'
+
+  return (
+    <DropdownMenu.Item
+      onClick={() => {
+        dispatch(goToViewRecordPage(declarationId as string))
+      }}
+    >
+      <Icon name="Eye" color="currentColor" size="large" />
+      {intl.formatMessage(messages.view, { recordOrDeclaration })}
+    </DropdownMenu.Item>
+  )
+}
+
+const CorrectRecordAction: React.FC<
+  IActionItemCommonProps & IDeclarationProps
+> = ({ declarationId, declarationStatus, type, isDownloaded, intl, scope }) => {
+  const dispatch = useDispatch()
+
+  const isBirthOrDeathEvent =
+    type && [Event.Birth, Event.Death].includes(type.toLowerCase() as Event)
```

Comment:
> type && [Event.Birth, Event.Death].includes(type.toLowerCase() as Event)

Lets compare the actual event types together. If it means refactoring the calling end - we just need to spend time to do that. We have had issues with losing type safety when treating the event enum's as strings and lowercasing them